executor/local/stage.go

```diff
@@ -6,26 +6,169 @@ package local
 
 import (
 	"context"
+	"fmt"
+	"strings"
+	"time"
 
+	"github.com/go-vela/types/constants"
 	"github.com/go-vela/types/pipeline"
 )
 
 // CreateStage prepares the stage for execution.
 func (c *client) CreateStage(ctx context.Context, s *pipeline.Stage) error {
+	// load the logs for the init step from the client
+	l, err := c.loadStepLogs(c.pipeline.Stages[0].Steps[0].ID)
+	if err != nil {
+		return err
+	}
+
+	// update the init log with progress
+	//
+	// https://pkg.go.dev/github.com/go-vela/types/library?tab=doc#Log.AppendData
+	l.AppendData([]byte(fmt.Sprintf("> Pulling step images for stage %s...\n", s.Name)))
+
+	// create the steps for the stage
+	for _, step := range s.Steps {
+		// TODO: make this not hardcoded
```

Comment:
Assuming this todo will be updated in a later PR?

Reply:
Yes you are correct 👍 

I plan to refactor chunks of the codebase once I have a functional version of the `local` executor